crates/gen-js/src/lib.rs

```diff
@@ -2488,6 +2495,64 @@ impl Js {
                 }
             "),
 
+
+            Intrinsic::ToInt32 => self.src.js("
+                export function to_int32(val) {
+                    return val >> 0;
+                }
+            "),
+            Intrinsic::ToUint32 => self.src.js("
+                export function to_uint32(val) {
+                    return val >>> 0;
+                }
+            "),
+
+            Intrinsic::ToInt16 => self.src.js("
+                export function to_int16(val) {
+                    val >>>= 0;
+                    val %= 2 ** 16;
+                    if (val >= 2 ** 15) {
+                        val -= 2 ** 16;
+                    }
+                    return val;
+                }
+            "),
+            Intrinsic::ToUint16 => self.src.js("
+                export function to_uint16(val) {
+                    val >>>= 0;
+                    val %= 2 ** 16;
+                    return val;
+                }
+            "),
+            Intrinsic::ToInt8 => self.src.js("
+                export function to_int8(val) {
+                    val >>>= 0;
+                    val %= 2 ** 8;
+                    if (val >= 2 ** 7) {
+                        val -= 2 ** 8;
+                    }
+                    return val;
+                }
+            "),
+            Intrinsic::ToUint8 => self.src.js("
+                export function to_uint8(val) {
+                    val >>>= 0;
+                    val %= 2 ** 8;
+                    return val;
+                }
+            "),
+
+            Intrinsic::ToBigInt64 => self.src.js("
+                export function to_int64(val) {
+                    return BigInt.asIntN(64, val);
```

Comment:
Similar to above it may be better to do this inline instead of as an intrinsic perhaps?